airbyte-integrations/connectors/source-posthog/source_posthog/streams.py

```diff
@@ -0,0 +1,306 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+import math
+import urllib.parse
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class PosthogStream(HttpStream, ABC):
+    url_base = "https://app.posthog.com/api/"
+    primary_key = "id"
+    data_field = "results"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        resp_json = response.json()
+        if resp_json.get("next"):
+            next_query_string = urllib.parse.urlsplit(resp_json["next"]).query
+            params = dict(urllib.parse.parse_qsl(next_query_string))
+            if params:
+                return params
+
+    def request_headers(self, **kwargs) -> Mapping[str, Any]:
+        return {"Content-Type": "application/json", "User-Agent": "posthog-python/1.4.0"}
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        yield from response_json.get(self.data_field)
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+
+        params = {}
+        if next_page_token:
+            params.update(next_page_token)
+        return params
+
+
+class IncrementalPosthogStream(PosthogStream, ABC):
+    state_checkpoint_interval = math.inf
```

Comment:
Why did you use `math.inf?`

Reply:
Since a lot of endpoints have no way to both sort+filter, usually I go from the newest value and until I reach the state value. Then I change a state to newest value. I cannot save intermediate state due to reasons above, and it looks infinity makes some sence.